src/core/inductive.ml

```diff
@@ -0,0 +1,64 @@
+(*
+   We assume we have:
+// Builtins
+<builtin> ::=
+  | "0"
+  | "+1"
+  | "T"
+  | "P"    pour la fonction interprétant Prop dans TYPE
+  | "S"    pour la fonction interprétant Set  dans TYPE
+  | "Prop" pour le type des propositions
+  | "Set"  pour le type des données
+  | "eq"
+  | "refl"
+  | "eq_ind"
+  | "top"
+  | "bot"
+  | "not"
+  | "or"
+  | "and"
+  | "imp"
+*)
+
+open Timed
+open Terms
+open Syntax
+
+type inductive =
+  { name  : ident      ; (* the name of the inductive type     *)
+    sort  : p_type     ; (* the sort of the inductive type     *)
+    rule  : sym list   ; (* the list of constructors           *)
+    _ind  : sym option ; (* one inductive principle (Prop one) *)
+    _rec  : sym option ; (* one recursive principle (Set one)  *)
+    _rect : sym option ; (* one recursive principle (Type one) *)
+    _inv  : sym option   (* one inversion principle            *) }
```

Comment:
Focus on induction principle in Prop only for the moment.

Reply:
Yes, don't worry :)